python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,417 @@
+
+        # Make sure that a new lfo was created
+        lfo = await self.electrometer.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def take_fiber_spectrum(
+        self,
+        delay: float,
+        integration_time: float,
+    ) -> salobj.type_hints.BaseMsgType:
+        """Wait, then start an acquisition with the fiber spectrograph.
+
+        By default, this method will wait for `delay` seconds then start
+        an acquisition with the fiber spectrograph. Optionally the user may
+        provide a coroutine that will be awaited before the delay starts.
+
+        Parameters
+        ----------
+        delay : `float`
+            Seconds to wait before starting fiber spectrograph acquisition.
+        integration_time : `float`
+            Integration time for the fiber spectrum (seconds).
+
+        Returns
+        -------
+        RuntimeError : If exposure or return of lfo doesn't occur.
+        """
+        self.fiberspectrograph.evt_largeFileObjectAvailable.flush()
+
+        try:
+            await self.fiberspectrograph.cmd_expose.set_start(
+                integrationTime=integration_time,
+                timeout=integration_time + self.long_timeout,
+            )
+        except salobj.AckTimeoutError:
+            self.log.exception("Timed out waiting for the command ack. Continuing.")
+
+        lfo = await self.fiberspectrograph.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def perform_flat(self, exptime_dict: dict[str, float]) -> None:
+        # check dictionary
+        # Get the image number
+        # Somehow need to make sure camera is setup
+        if self.atcamera is not None:
+            asyncio.gather(
+                self.electrometer_scan(duration=exptime_dict["electrometer"]),
+                self.take_fiber_spectrum(
+                    delay=self.delay, integration_time=exptime_dict["fiberspectrograph"]
+                ),
+                self.atcamera.take_flats(exptime=exptime_dict["camera"], nflats=1),
+            )
+
+    def _get_monochromator_setup(
+        self,
+        filter: typing.Union[int, str, None],
+        calib_type: typing.Union[int, str, None],
+    ) -> typing.Tuple[int, float, float]:
+        """Get parameters for atmonochromator as function of the filter"""
+        if calib_type == "White":
+            grating = 3  # "Mirror"
+            exit_slit = 7.0
+            entrance_slit = 7.0
+        elif calib_type == "Mono":
+            grating = 0  # Need to figure out how to make these come
+            # from a configuration self.filter_gratings[filter]
+            exit_slit = 0.5
+            entrance_slit = 0.5
+
+        return grating, entrance_slit, exit_slit
+
+    async def get_optimized_exposure_times(
+        self, calib_type: str, filter: str, wavelength: float
```

Comment:
See my comment above about `calib_type`.

Reply:
Not sure if I've done this correctly